### ISSUE TITLE
feature: helm pause

### ISSUE THREAD

**Issue by Suozz** [helm/helm#10394]
can helm pause a upgrade

**Comment by hickeyma** [comment]
@Suozz Can you provide more details for the feature and give examples of use cases?

**Comment by Suozz** [comment]
> @Suozz Can you provide more details for the feature and give examples of use cases?

My deployment has 2000 deployments, I want to pause my deployment when upgrade my deployment by helm

**Comment by yxxhero** [comment]
@Suozz I think this is not a function that helm should provide, you should start with the features of k8s.

**Comment by bacongobbler** [comment]
Helm is what is known as a declarative package manager. Declarative programs describe their desired results without explicitly listing commands or steps required, meaning that the programmer has no control over the exact procedure to be followed.

The Kubernetes API is fundamentally declarative, which means that the controllers always work to reconcile the observed state with the desired state. Therefore, if you ask Kubernetes (via Helm) to upgrade a Deployment containing 2000 pods, the ReplicaSet controller will replace the existing pods with new ones to match the desired state. There is no way to pause a rollout in progress as the controller is just trying to match the desired state.

In other words, this is not something we will support as it goes against the design principles of a declarative package manager (and Kubernetes for that matter). If a fault occurs in your upgrade process, inspect the status of the upgrade and make changes to the desired state as required.

Closing this issue as answered. Please feel free to comment here and let us know if you have further questions. Thanks!

**Comment by zachwalton** [comment]
Not saying Helm should necessarily provide this (though I googled this because I hoped it did), but:

> There is no way to pause a rollout in progress

> goes against the design principles of ... Kubernetes

https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#pausing-and-resuming-a-deployment
